command-expansion-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -2,46 +2,122 @@
 export class Command<A extends ArgType[] | { [argName: string]: any } = [] | {}> {
   public readonly stem: string;
   public readonly arguments: A;
+  public readonly absoluteTime?: Temporal.Duration;
+  public readonly relativeTime?: Temporal.Duration;
+  public readonly epochTime?: { epochName: string; time: Temporal.Duration };
 
-  private constructor(opts: { stem: string; arguments: A }) {
+  private constructor(opts: {
+    stem: string;
+    arguments: A;
+    absoluteTime?: Temporal.Duration;
+    relativeTime?: Temporal.Duration;
+    epochTime?: { epochName: string; time: Temporal.Duration };
+  }) {
     this.stem = opts.stem;
     this.arguments = opts.arguments;
+    this.absoluteTime = opts.absoluteTime;
```

Comment:
We need to prevent multiple of these from being set on the same command.

Reply:
I am not sure I understand. The command will expand to the latest time value you specify

```
PEEL_BANANA.absoluteTiming(Temporal.Duration.from({ minutes: 10 })).absoluteTiming(Temporal.Duration.from({ minutes: 50 })),

BAKE_BREAD.absoluteTiming(Temporal.Duration.from({ minutes: 10 })).epochTiming("Saturn", Temporal.Duration.from({ minutes: 15 }))
```

PEEL_BANANA - 50 minutes absolute
BAKE_BREAD - 15 minutes epoch

Reply:
Yes, but we aren't enforcing in the type system that only one of these values is populated at once. We can do this by typing the input type as
```
{
  stem: string;
  arguments: A;
} & (
  {
    absoluteTime: Temporal.Duration;
    epochTime: null;
    relativeTime: null;
  }
  | {
    absoluteTime: null;
    epochTime: { epochName: string; time: Temporal.Duration };
    relativeTime: null;
  }
  | {
    absoluteTime: null;
    epochTime: null;
    relativeTime: Temporal.Duration;
  }
)
```

Reply:
Basically keeping to the rule of "Make invalid states unrepresentable"

Reply:
I did some reading up on this as I was a little confusing when looking at the example you provided. I have never seen intersection types before. I got it implemented now. 

https://www.typescriptlang.org/docs/handbook/2/objects.html#intersection-types

The change required me to update [CommandTypeCodegen.ts](https://github.com/NASA-AMMOS/aerie/pull/149/commits/09058f7ed1a2fc7d28df2ff2a97c8c37e286b08b#diff-664406c819e249960325a4124e84cd5c9f675976870b5a8e5c7a92ad6cf9ea73)